packages/ember-cli-fastboot/fastboot/initializers/ajax.js

```diff
@@ -30,7 +31,10 @@ export default {
 
   initialize: function(application) {
     application.register('ajax:node', nodeAjax, { instantiate: false });
-    application.inject('adapter', '_ajaxRequest', 'ajax:node');
-    application.inject('adapter', 'fastboot', 'service:fastboot');
+
+    if (!gte('4.0.0')) {
+      application.inject('adapter', '_ajaxRequest', 'ajax:node');
+      application.inject('adapter', 'fastboot', 'service:fastboot');
+    }
```

Comment:
I don't think this check is needed, and these lines can just be removed. This would have to be released as a breaking change though.

That way apps using versions of Ember prior to 4.x would be able to use this without the implicit injections, to make the transition to 4.x easier. That's what we're running into where I work - we want to get rid of the deprecations on 3.28 before moving to 4.x.

Reply:
It's also an odd behavior to have a version that both does and doesn't inject based on the Ember version. I agree that removing them and cutting a new major is the way to go